gatox/github/api.py

```diff
@@ -1304,26 +1304,19 @@ async def retrieve_workflow_ymls_ref(self, repo_name: str, ref: str):
         """
         ymls = []
 
-        page = 1
-        per_page = 100
         objects = []
 
-        while True:
-            resp = await self.call_get(
-                f"/repos/{repo_name}/contents/.github/workflows/",
-                params={"ref": ref, "per_page": per_page, "page": page},
-            )
-
-            if resp.status_code != 200:
-                break
-
-            listing = resp.json()
-            objects.extend(listing)
-
-            if len(listing) < per_page:
-                break
+        resp = await self.call_get(
+            f"/repos/{repo_name}/contents/.github/workflows",
+            params={"ref": ref},
+        )
 
-            page += 1
+        if resp.status_code == 200:
+            objects = resp.json()
+        else:
+            logger.warning(
+                f"Failed to retrieve workflows from {repo_name} at ref {ref}!"
+            )
```

Comment:
Not related to the PR but found a potential improvement, for this block:

```
results = await asyncio.gather(*tasks, return_exceptions=True)
ymls = [wf for wf in results if wf is not None]
```

`results` could contain a mix of `Exception`and `Workflow` items. Might be worth checking if there's any and raising/logging the error